server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/DomainQuotaRoutesTest.java

```diff
@@ -60,6 +61,73 @@ void setUp(WebAdminQuotaSearchTestSystem testSystem) throws Exception {
         RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
     }
 
+    @Test
+    void getShouldBeAuthenticated(WebAdminQuotaSearchTestSystem testSystem) {
+        with()
+            .get(QUOTA_DOMAINS + "/" + PERDU_COM);
+
+        assertThat(testSystem.getAuthenticationFilter().hasBeenAuthenticated()).isTrue();
+    }
+
+    @Test
+    void getCountShouldBeAuthenticated(WebAdminQuotaSearchTestSystem testSystem) {
+        with()
+            .get(QUOTA_DOMAINS + "/" + PERDU_COM + "/" + COUNT);
+
+        assertThat(testSystem.getAuthenticationFilter().hasBeenAuthenticated()).isTrue();
+    }
+
+    @Test
+    void getSizeShouldBeAuthenticated(WebAdminQuotaSearchTestSystem testSystem) {
+        with()
+            .get(QUOTA_DOMAINS + "/" + PERDU_COM + "/" + SIZE);
+
+        assertThat(testSystem.getAuthenticationFilter().hasBeenAuthenticated()).isTrue();
+    }
+
+    @Test
+    void deleteCountShouldBeAuthenticated(WebAdminQuotaSearchTestSystem testSystem) {
+        with()
+            .delete(QUOTA_DOMAINS + "/" + PERDU_COM + "/" + COUNT);
+
+        assertThat(testSystem.getAuthenticationFilter().hasBeenAuthenticated()).isTrue();
+    }
+
+    @Test
+    void deleteSizeShouldBeAuthenticated(WebAdminQuotaSearchTestSystem testSystem) {
+        with()
+            .delete(QUOTA_DOMAINS + "/" + PERDU_COM + "/" + SIZE);
+
+        assertThat(testSystem.getAuthenticationFilter().hasBeenAuthenticated()).isTrue();
+    }
+
+    @Test
+    void putCountShouldBeAuthenticated(WebAdminQuotaSearchTestSystem testSystem) {
+        with()
+            .body("42")
+            .put(QUOTA_DOMAINS + "/" + PERDU_COM + "/" + COUNT);
+
+        assertThat(testSystem.getAuthenticationFilter().hasBeenAuthenticated()).isTrue();
```

Comment:
usually, we just try without authentication and check that return code is 401

Reply:
Agree. JWT set up in WebAdmin route tests is a must.

Reply:
Then the downside is that our testing code do not act only on the HTTP layer, and thus, is less portable.

Reply:
Yes, but you are not locked on a bounded authentication filter; i.e. changing this filter will break everything.